src/faber/tools/python.py

```diff
@@ -72,23 +72,42 @@ def __init__(self, name='python', command=None, version='', features=()):
         # Now determine all the flags we may need to compile C / C++ extensions
         self.include = include(self.check_sysconfig('get_python_inc()'))
         platform = self.check_python('import platform; print(platform.system())')
+        impl = self.check_python('import platform; print(platform.python_implementation())')
+        prefix = self.check_python('import sys; print(sys.prefix)')
         if platform == 'Windows':
-            version = self.check_python('import sys; print("%d%d"%sys.version_info[0:2])')
-            prefix = self.check_python('import sys; print(sys.prefix)')
-            self.libfile = join(prefix, 'libs', 'python{}.lib'.format(version))
-            self.libpath = join(prefix, 'libs')
-            self.lib = 'python' + version
+            if impl == 'CPython':
+                version = self.check_python('import sys; print("%d%d"%sys.version_info[0:2])')
+                self.libfile = join(prefix, 'libs', 'python{}.lib'.format(version))
+                self.libpath = join(prefix, 'libs')
+                self.lib = 'python' + version
+            elif impl == 'PyPy':
+                version = self.check_python('import sys; print(sys.version_info[0])')
+                self.libfile = join(prefix, 'libs', 'libpypy{}-c.lib'.format(version))
+                self.libpath = join(prefix, 'libs')
+                self.lib = 'libpypy' + version + '-c'
+            else:
+                raise ValueError('unsupported Python implementation')
         else:
             self.libpath = self.check_sysconfig('get_config_var("LIBDIR")')
-            self.libfile = self.check_sysconfig('get_config_var("LIBRARY")')
-            match = re.search(r'(python.*)\.(a|so|dylib)', self.libfile)
-            if match:
-                self.lib = match.group(1)
-                if match.group(2) == 'a':
-                    flags = self.check_sysconfig('get_config_var("LINKFORSHARED")')
-                    if flags is not None:
-                        flags=flags.split()
-                        self.ldflags = ldflags(*flags)  # TODO: use them !
+            if impl == 'CPython':
+                self.libfile = self.check_sysconfig('get_config_var("LIBRARY")')
+                match = re.search(r'(python.*)\.(a|so|dylib)', self.libfile)
+                if match:
+                    self.lib = match.group(1)
+                    if match.group(2) == 'a':
+                        flags = self.check_sysconfig('get_config_var("LINKFORSHARED")')
+                        if flags is not None:
+                            flags=flags.split()
+                            self.ldflags = ldflags(*flags)  # TODO: use them !
```

Comment:
does this need an `else` clause?

Reply:
Perhaps, but as this code is merely shifting these lines around, I'd rather not touch the logic within this PR.